packages/plugins/model-evaluate/bayesian-model-evaluate/src/script.ts

```diff
@@ -0,0 +1,160 @@
+    [key: string]: [ string, number ];
+  }
+  const all_words_dict: AllWordsDict = {};
+  for (const word_list of train_data_list) {
+    for (const word of word_list) {
+      if (!all_words_dict[word]) {
+        all_words_dict[word] = [ word, 1 ];
+      } else {
+        all_words_dict[word][1] += 1;
+      }
+    }
+  }
+
+  const all_words_tuple_list = Object.values(all_words_dict).sort((item1, item2) => item2[1] - item1[1]);
+  const all_words_list = unzip(all_words_tuple_list)[0];
+
+  return [ all_words_list, train_data_list, train_class_list ];
+};
+
+function words_dict(all_words_list: string[], stopwords_set = new Set<string>()): string[] {
+  const feature_words: string[] = [];
+  for (let word of all_words_list) {
+    if (!isdigit(word) &&
+        !stopwords_set.has(word) &&
+        word.length > 1 &&
+        word.length < 5) {
+      feature_words.push(word);
+    }
+  }
+
+  return feature_words;
+}
+
+export const TextFeatures = function(train_data_list: string[], feature_words: string[]) {
+  function text_features(text: string, feature_words: string[]) {
+    const text_words = new Set(text);
+    const features = [];
+    for (const word of feature_words) {
+      if (text_words.has(word)) {
+        features.push(1);
+      } else {
+        features.push(0);
+      }
+    }
+
+    return features;
+  }
+
+  return train_data_list.map((text: string) => {
+    return text_features(text, feature_words);
+  });
+};
+
+export const processPredictData = function (data: any, all_words_list_path: string, stopwords_path: string): Promise<Set<string>> {
+  const all_words_list = pickle.load(fs.readFileSync(all_words_list_path));
+  const word_cut = jieba.cut(data, boa.kwargs({
+    cut_all: false
+  }));
+
+  const stopwords_file = stopwords_path;
+  const text_features = function (text: string, feature_words: string[]): number[] {
+    const text_words = new Set(text);
+    const features = [];
+    for (let word of feature_words) {
+      if (text_words.has(word)) {
+        features.push(1);
+      } else {
+        features.push(0);
+      }
+    }
+
+    return features;
+  };
+
+  return MakeWordsSet(stopwords_file).then((stopwords_set) => {
+    const feature_words = words_dict(all_words_list, stopwords_set);
+    return word_cut.map((text: string) => {
+      return text_features(text, feature_words);
+    });
+  });
+};
+
+export const save_all_words_list = function(feature_words: any, filepath: string) {
+  pickle.dump(feature_words, fs.readFileSync(filepath));
+};
+
+export const get_all_words_list = function(filepath: string) {
+  return pickle.load(fs.readFileSync(filepath));
+};
+
+export const getBayesModel = function() {
+  return MultinomialNB();
+};
+
+export const saveBayesModel = function(classifier: any, filepath: string) {
+  pickle.dump(classifier, fs.readFileSync(filepath));
+};
+
+export const loadModel = function (filepath: string) {
+  return pickle.load(fs.readFileSync(filepath));
```

Comment:
As @utkobe discussed with others, this needs Python's "open", because the pickle.load receive the Python's file handle, see https://docs.python.org/3/library/pickle.html#pickle.load.

Reply:
In #107 , 'open' was be replaced :(

Reply:
Yea, that's not working and we'll fix it later.

Reply:
OO